Realm/Realm/DatabaseTypes/Accessors/ManagedAccessor.cs

```diff
@@ -181,12 +181,15 @@ public void SubscribeForNotifications(Action<string> notifyPropertyChangedDelega
                 throw new RealmFrozenException("It is not possible to add a change listener to a frozen RealmObjectBase since it never changes.");
             }
 
+            PropertyType collectionFlag = PropertyType.Array | PropertyType.Dictionary | PropertyType.Set;
+            List<IntPtr> propertyIndices = (from p in ObjectSchema where (p.Type & collectionFlag) == 0 select Metadata.GetPropertyIndex(p.Name, Metadata)).ToList();
```

Comment:
```suggestion
            IntPtr[] propertyIndices = (from p in ObjectSchema where (p.Type & collectionFlag) == 0 select Metadata.GetPropertyIndex(p.Name, Metadata)).ToArray();
```
We don't need a mutable list here, an array is enough. And yes, they can be cached like @papafe says.